app/lib/dfe/reference_data/joined_reference_list.rb

```diff
@@ -0,0 +1,45 @@
+module DfE
+  module ReferenceData
+    ##
+    #
+    # A +JoinedReferenceList+ is a wrapper around one or more +ReferenceList+s with
+    # disjoint (non-overlapping) ID ranges, which joins them all together into one
+    # big one.
+    class JoinedReferenceList < ReferenceList
```

Comment:
Doesn't look like this class is used at the moment. Could we leave it out for now?

Reply:
I thought of that possibility but the next work after this is done is to extract the `app/lib/dfe` to its own gem so I left the original code to make that work easy. Wdyt? cc @stevehook  / @duncanjbrown

Reply:
We incorporated this and the above as they were in Alaric's original design... but I do agree with @stevehook that if this stuff isn't used it's better not to include it. @alaric-dfe would you be happy for us to leave this out now and add it to the gem as and when it's needed?

Reply:
Sure! I put `Tweaked` and `Joined` in largely to show that they were possible, as they're part of the customisation mechanism that individual services can use to make their own "variations" on the lists - see https://github.com/DFE-Digital/apply-for-teacher-training/blob/4342-implement-a-new-shared-source-of-truth-or-at-least-copy-and-pasteable-for-degree-subjects-institutions-and-types/app/lib/dfe/reference_data/degrees.rb#L627 for an example.

Reply:
Ok, 🚀 ! I'll delete the `TweakedReferenceList` only because the `JoinedReferenceList` is used on the degrees types including the generics one.